Source/Scene/ModelInstanceCollection.js

```diff
@@ -0,0 +1,844 @@
+        this._createVertexBuffer = true;
+        this._vertexBufferDirty = false;
+        this._instancedUniformsByProgram = undefined;
+
+        this._drawCommands = [];
+        this._pickCommands = [];
+        this._modelCommands = undefined;
+
+        this._boundingVolume = options.boundingVolume;
+        this._boundingVolumeModel = new Matrix4();
+        this._boundingVolumeModelView = new Matrix4();
+
+        // Passed on to Model
+        this._url = options.url;
+        this._headers = options.headers;
+        this._gltf = options.gltf;
+        this._basePath = options.basePath;
+        this._cacheKey = options.cacheKey;
+        this._asynchronous = options.asynchronous;
+
+        /**
+         * This property is for debugging only; it is not for production use nor is it optimized.
+         * <p>
+         * Draws the bounding sphere for each draw command in the model.  A glTF primitive corresponds
+         * to one draw command.  A glTF mesh has an array of primitives, often of length one.
+         * </p>
+         *
+         * @type {Boolean}
+         *
+         * @default false
+         */
+        this.debugShowBoundingVolume = defaultValue(options.debugShowBoundingVolume, false);
+        this._debugShowBoundingVolume = false;
+
+        /**
+         * This property is for debugging only; it is not for production use nor is it optimized.
+         * <p>
+         * Draws the model in wireframe.
+         * </p>
+         *
+         * @type {Boolean}
+         *
+         * @default false
+         */
+        this.debugWireframe = defaultValue(options.debugWireframe, false);
+        this._debugWireframe = false;
+    };
+
+    defineProperties(ModelInstanceCollection.prototype, {
+        show : {
+            get : function() {
+                return this._show;
+            }
+        },
+        allowPicking : {
+            get : function() {
+                return this._allowPicking;
+            }
+        },
+        length : {
+            get : function() {
+                return this._instances.length;
+            }
+        },
+        ready : {
+            get : function() {
+                return this._ready;
+            }
+        },
+        readyPromise : {
+            get : function() {
+                return this._readyPromise;
+            }
+        }
+    });
+
+    ModelInstanceCollection.prototype.getModel = function(index) {
+        //>>includeStart('debug', pragmas.debug);
+        if (!defined(index) || (index < 0) || (index >= this.length)) {
+            throw new DeveloperError('index is required and between zero and length - 1 (' + (this.length - 1) + ').');
+        }
+        //>>includeEnd('debug');
+
+        return this._instances[index];
+    };
+
+    ModelInstanceCollection.prototype._updateInstance = function(instance) {
+        // TODO : update the whole buffer for now, but later do sub-commits based on the instance's index
```

Comment:
This could be a `PERFORMANCE_TODO`.  It is a good idea, but probably not something we'll do soon.